cmd/zpool/zpool.d/smart

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+#
+# Show SMART stats
+#
+
+helpstr="
+smart:		Show SMART temperature and important error stats (specific to drive type)
+smartx:		Show SMART extended drive stats (specific to drive type).
+sm_temp:	Show SMART drive temperature in celsius (all drives).
+sm_status:	Show reported SMART status (all drives).
+sm_rproc:	Show SMART read GBytes processed over drive lifetime (SAS).
+sm_wproc:	Show SMART write GBytes processed over drive lifetime (SAS).
+sm_rucor:	Show SMART read uncorrectable errors (SAS).
+sm_wucor:	Show SMART write uncorrectable errors (SAS).
+sm_nonmed:	Show SMART non-medium errors (SAS).
+sm_defect:	Show SMART grown defect list (SAS).
+sm_hours:	Show number of hours drive powered on (all drives).
+sm_realloc:	Show SMART reallocated sectors count (ATA).
+sm_rep_ucor:	Show SMART reported uncorrectable count (ATA).
+sm_cmd_to:	Show SMART command timeout count (ATA).
+sm_pend_sec:	Show SMART current pending sector count (ATA).
+sm_off_ucor:	Show SMART offline uncorrectable errors (ATA).
+sm_ata_err:	Show SMART ATA errors (ATA).
+sm_pwr_cyc:	Show SMART power cycle count (ATA).
+"
+
+script=$(basename "$0")
+
+if [ "$1" = "-h" ] ; then
+        echo "$helpstr" | grep "$script:" | tr -s '\t' | cut -f 2-
+        exit
+fi
+
+
+# Are we a SAS or ATA drive?  Look for the right line in smartctl:
+#
+# SAS:
+#	Transport protocol:   SAS
+#
+# SATA:
+#	ATA Version is:   8
+#
+
+raw_out=$(eval "sudo smartctl -a $VDEV_UPATH")
```

Comment:
Drop the `sudo` here.  We already provide a /etc/sudoers one-liner in the zpool man page to encourage people to allow normal users to run smartctl:

```
For example, to allow user "zfsuser" access to "smartctl -a", add
the following to /etc/sudoers:

zfsuser ALL=NOPASSWD: /usr/sbin/smartctl -a /dev/sd[a-z]*, NOEXEC: /usr/sbin/smartctl -a /dev/sd[a-z]*```